### PR TITLE
feat: add type define and schema for config

### PR DESCRIPTION
**Pull request by targeral**
# PR Details

add tpe define and schema for config

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/solutions/module-tools-v2/src/types/config/index.ts [hunk 1]
```diff
@@ -1,44 +1,154 @@
...
+  target: Target;
+  entry: Entry;
+  dts: DTS;
+  sourceMap: SourceMap;
+  copy: Copy;
+}
+
+export type BundleFormat = 'esm' | 'cjs' | 'umd' | 'iife';
+export type BundleOptions = {
+  platform: any;
```
Comment (chenjiahan) [inline packages/solutions/module-tools-v2/src/types/config/index.ts:40]: 有一些 any 喔，是准备后面再补全嘛

File: packages/solutions/module-tools-v2/src/config/normalize.ts [hunk 2]
```diff
@@ -2,7 +2,9 @@ import type { PluginAPI } from '@modern-js/core';
 import type { UserConfig, BaseBuildConfig } from '../types';
 
 export const normalizeBuildConfig = (api: PluginAPI): BaseBuildConfig[] => {
-  const { buildConfig = [] } =
+  const { buildConfig } =
     api.useResolvedConfigContext() as unknown as UserConfig;
-  return buildConfig as BaseBuildConfig[];
+  return (
+    Array.isArray(buildConfig) ? buildConfig : [buildConfig]
```
Comment (chenjiahan) [inline packages/solutions/module-tools-v2/src/config/normalize.ts:8]: 这里可以使用 ensureArray 这个 utils

File: packages/solutions/module-tools-v2/src/types/config/index.ts [hunk 3]
```diff
@@ -1,44 +1,154 @@
...
+  UserConfig as UserConfig_,
+  ToolsConfig as ToolsConfig_,
+  NewPluginConfig,
+} from '@modern-js/core';
+import { ModuleToolsHooks } from '..';
+import type { DeepPartial } from '../utils';
 
-export interface BaseBuildConfig {
-  buildType: 'bundle' | 'bundleless';
+export type Target =
```
Comment (chenjiahan) [inline packages/solutions/module-tools-v2/src/types/config/index.ts:9]: libuild 里会有 Target 类型定义吗，如果两者是一样的，最好从 libuild 里导入

File: packages/solutions/module-tools-v2/src/types/config/index.ts (hunk 1 above)
Comment (targeral) [reply packages/solutions/module-tools-v2/src/types/config/index.ts:40]: 是的，这块目前还没有确定。后面会补充这部分 any 的类型

File: packages/solutions/module-tools-v2/src/types/config/index.ts (hunk 3 above)
Comment (targeral) [reply packages/solutions/module-tools-v2/src/types/config/index.ts:9]: 可以，后面等接入的时候，我确认一下。

File: packages/solutions/module-tools-v2/src/config/normalize.ts (hunk 2 above)
Comment (targeral) [reply packages/solutions/module-tools-v2/src/config/normalize.ts:8]: 等开发这部分逻辑的时候，我改一下。这块目前是稍作处理

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**